### PR TITLE
WIP: more changes needed to get lbryum to work on regtest

### PR DESCRIPTION
**Pull request by jobevers**

### CODE REVIEW

**Comment by jobevers** [comment]
@kaykurokawa 

I tried running lbryum on regtest yesterday and ran into issues.  I had an incredibly difficult time understanding the lbryum code and how it mapped to the lbrycrd code so I re-worked a lot of the `get_target` calculation to follow the lbrycrd code much more carefully.

I need to clean up the commits, but would appreciate it you could look this over and give some initial feedback before I put more work into it.

Thanks.

File: lib/blockchain.py [hunk 1]
```diff
@@ -16,21 +16,29 @@
...
-MAX_TARGET = 0x0000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
 HEADER_SIZE = 112
 BLOCKS_PER_CHUNK = 96
 
 HEADERS_URL = "https://s3.amazonaws.com/lbry-blockchain-headers/blockchain_headers_latest"
 
-class Blockchain(util.PrintError):
-    '''Manages blockchain headers and their verification'''
 
+class ChainValidationError(Exception):
```
Comment (jobevers) [inline lib/blockchain.py]: this should get taken out.  I was going to replace all the AssertionErrors with a more specific error statement so that the blanket try/catches could be made more specific.

File: lib/blockchain.py [hunk 2]
```diff
@@ -181,48 +190,42 @@ def read_header(self, block_height):
...
-        if chain == 'regtest':
-            nTargetTimespan = 1
-        else:
-            nTargetTimespan = 150
+        nTargetTimespan = self.N_TARGET_TIMESPAN
+        log.info('Using target timespan of %s', nTargetTimespan)
         nModulatedTimespan = nTargetTimespan - (nActualTimespan - nTargetTimespan) / 8
         nMinTimespan = nTargetTimespan - (nTargetTimespan / 8)
         nMaxTimespan = nTargetTimespan + (nTargetTimespan / 2)
-        nModulatedTimespan = nMinTimespan if nModulatedTimespan < nMinTimespan else nMaxTimespan
```
Comment (jobevers) [inline lib/blockchain.py]: The logic changed on this.  The original code seems wrong.

File: lib/blockchain.py [hunk 3]
```diff
@@ -271,3 +274,120 @@ def connect_chunk(self, idx, hexdata):
...
+    def check_bits(self, bits):
+        bitsN = (bits >> 24) & 0xff
+        assert 0x03 <= bitsN <= 0x1f, \
+            "First part of bits should be in [0x03, 0x1d], but it was {}".format(hex(bitsN))
+        bitsBase = bits & 0xffffff
+        assert 0x8000 <= bitsBase <= 0x7fffff, \
+            "Second part of bits should be in [0x8000, 0x7fffff] but it was {}".format(bitsBase)
+
+
+class LbryCrdTest(_Blockchain):
```
Comment (jobevers) [inline lib/blockchain.py]: I haven't been able to test the LbryCrdTest blockchain

File: lib/simple_config.py [hunk 4]
```diff
@@ -208,20 +208,17 @@ def read_user_config(path):
...
         result = json.loads(data)
-    except:
+    except Exception:
         try:
             result = ast.literal_eval(data)
-        except:
-            #print_msg("Error: Cannot read config file.")
-            return {}
+        except Exception:
+            raise Exception('Failed to parse config file {}'.format(config_path))
```
Comment (jobevers) [inline lib/simple_config.py]: I think it is insane that electrum / lbryum simple swallow errors while reading the config file.

This should be a separate commit though.

**Comment by kaykurokawa** [comment]
Didn't test anything but I think in concept these are all good changes.

**Comment by kaykurokawa** [comment]
Closing as replaced by https://github.com/lbryio/lbryum/pull/63 and https://github.com/lbryio/lbryum/pull/62
